lib/api/controllers/funnelController.js

```diff
@@ -127,7 +131,10 @@ class FunnelController {
 
     // resolves the callback immediately if a slot is available
     if (this.concurrentRequests < this.kuzzle.config.limits.concurrentRequests) {
-      return callback(null);
+      if (this.cachedIds[request.id]) {
+        delete this.cachedIds[request.id];
```

Comment:
Won't this leave the request in the cache queue, making the funnel processing it twice?
Once here and now if the request slot is immediately available, and once later by the `playCachedRequests` method?